main.js

```diff
@@ -0,0 +1,121 @@
+const app = {
+	apiKey: '35545a9a120d7f1193670e2f13d3c4bb',
+	units: 'imperial',
+	lang: 'en',
+	init: () => {
+	  document
+		.getElementById('btnSearch')
+		.addEventListener('click', () => app.fetchWeather());
+	  document
+		.getElementById('btnLocation')
+		.addEventListener('click', app.getLocation);
+	},
+	getLocation: () => {
+	  if (navigator.geolocation) {
+		navigator.geolocation.getCurrentPosition(
+		  (position) => {
+			const { latitude: lat, longitude: lon } = position.coords;
+			app.fetchWeather({ lat, lon });
+		  },
+		  (error) => {
+			console.error(error);
+			alert('Unable to get location. Please allow location access.');
```

Comment:
good start, but a step above is displaying the error:
something like this
```
<div class="alert alert-danger" role="alert">
  Unable to fetch weather data. Please try again.
</div>
```

Reply:
great input, thank you!!